pandas/tests/io/parser/test_common.py

```diff
@@ -1927,3 +1927,13 @@ def test_read_table_deprecated(all_parsers):
                                     check_stacklevel=False):
         result = parser.read_table(StringIO(data))
         tm.assert_frame_equal(result, expected)
+
+
+def test_first_row_bom(all_parsers):
+    # see gh-26545
+    parser = all_parsers
+    data = '''\ufeff"Head1"	"Head2"	"Head3"'''
+
+    result = parser.read_csv(StringIO(data), delimiter='\t')
+    expected = DataFrame(columns=["Head1", "Head2", "Head3"])
+    tm.assert_frame_equal(result, expected)
```

Comment:
This test is fine to add.  `tm.assert_frame_equal` does the trick.

Reply:
> @LuckyDenis could you also check what `test_utf8_bom` is doing in the test module? It looks to cover the same intention as test added so want to make sure we understand the difference and clarify accordingly

@WillAyd, I looked at the difference between the tests. In the `test_utf8_bom`, the dataset only checks that it clears the first header from the bom and does not reveal the bug I fixed. My test checks just processing headers and that you grow them correctly.

Reply:
Awesome thanks for verifying that!